src/inmanta/data/__init__.py

```diff
@@ -2043,6 +2043,11 @@ class Project(BaseDocument):
     def to_dto(self) -> m.Project:
         return m.Project(id=self.id, name=self.name, environments=[])
 
+    async def delete(self, connection: Optional[asyncpg.connection.Connection] = None) -> None:
+        # flush the environment cache
+        Environment.flush_cache()
```

Comment:
Should we put the cache behind a lock? As in, prevent concurrent read-write, and hold the lock until `delete` is done?